### ISSUE TITLE
QGIS hanging whenever I move away from the program

### ISSUE THREAD

**Issue by patrickharvey333** [qgis/QGIS#41176]
**Describe the bug**

Recently purchased a new system and now I am encountering an issue with the program hanging every time I move to another application then come back to QGIS, or leave the system idle for more than a minute or two. Don't need to have a dataset loaded for this to happen either. It is a clean install and I have used a clean user profile. I uninstalled QGIS and reinstalled after using a registry cleaner and removed all plugins thinking they may be the issue. The problem still persists.

I have a feeling this will be something user specific for me and be associated with my new system. I have checked forums already and am at a loss as to what my next step should be. 

The attached image was the program hanging when I switched from my browser back to QGIS

![QGIS Hanging](https://user-images.githubusercontent.com/39748353/105767103-c3459e80-5fae-11eb-968a-87e3e9cd6068.png)

QGIS version: 3.16.3
OS version: Windows 10 Pro

PC Specs:
Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz   3.70 GHz
64.0 GB RAM
NVidia RTX2080S
Dual 2TB m2 SSD + 4TB SATA SSD

**Comment by roya0045** [comment]
Could be a qt issue also I would wager.

**Comment by patrickharvey333** [comment]
Thanks @roya0045. How would I resolve that?

**Comment by roya0045** [comment]
You could check if qt designer behaves the same way (if you have it from the install). Otherwise I'm not too sure what may be the issue.

**Comment by patrickharvey333** [comment]
Checked it out and it isn't behaving the same way.

**Comment by roya0045** [comment]
Can you try qgis alone with no other program active, just the barest minimum to make sure nothing is blocking (antivirus, etc) or interacting weirdly (graphic 'boosting' programs) with QGIS?

**Comment by patrickharvey333** [comment]
Gave it a try. No luck unfortunately. Also tried uninstalling some of my drivers (video, audio), cleaning the system then reinstalled them. With or without the drivers, I still had no luck. Did a complete clean of QGIS again but also no luck.

Thanks for the quick responses @roya0045

**Comment by patrickharvey333** [comment]
I took extreme measures and reinstalled windows. Working perfectly now.

**Comment by roya0045** [comment]
@patrickharvey333 sad to hear that it came to this but I'm glad it fixed your issues!

**Comment by patrickharvey333** [comment]
This seems to have started again for me. I know at least one other person that has been having issues with this too.

**Comment by patrickharvey333** [comment]
So to add to this...

Everything was running perfectly two days ago. I haven't installed anything new (drivers or apps). I have the latest version of QGIS (3.16.3)

any advice would be greatly appreciated.

**Comment by gioman** [comment]
> any advice would be greatly appreciated.

@patrickharvey333 seems very likely to be a local issue rather than a QGIS one.

**Comment by patrickharvey333** [comment]
Yeah, I figured that would be the case. What information can I get to someone that may be able to help? error reports? 

I don't really want to reinstall windows again to solve the problem.

**Comment by gioman** [comment]
I would try installing QGIS on a Windows virtual machine (you can download the Windows 10 iso from the MS site), and do not install anything else you would normally install on your real machine.

**Comment by patrickharvey333** [comment]
Thanks! I think I have worked out what is happening and where the conflict was that was causing the issue. 

I thought it was some kind of conflict between one of the Intel-HD drivers (there are several on my new system) and the Nvidia driver - it started with the fresh install post installation of the Intel drivers. I refreshed windows and only installed the Nvidia drivers. This seems to have fixed the issue. If that changes I will go to the virtual machine option and only run QGIS there. 

Thanks for all the help! I really appreciate it. :)
